azure-pipelines.yml

```diff
@@ -18,6 +18,26 @@ stages:
 - stage: InitialTests
   jobs:
 
+  - job: Lint
+    condition: and(succeeded(), ne(variables['Build.SourceBranch'], 'refs/heads/master'))  # skip for PR merges
+    pool:
+      vmImage: 'ubuntu-18.04'
+    steps:
+    - task: UsePythonVersion@0
+      inputs:
+        versionSpec: '3.8'
+        addToPath: true
+        architecture: 'x64'
+    - script: >-
+        python -m pip install
+        pycodestyle==2.5.0
+        GitPython==3.1.13
+      displayName: 'Install tools'
+      failOnStderr: true
+    - script: |
+        python tools/linter.py --branch origin/$(System.PullRequest.TargetBranch)
```

Comment:
This should be source :)

Reply:
ok so we merge and create a branch like `refs/remotes/pull/18423/merge`, Hmm, so is there a clean way to get this? Or I'll rewrite the logic to use the current branch we checkout as default. Not too hard as it's already checked out and scipy does that.

Reply:
Yes, doing the same as what SciPy does is nice - easier to maintain if it's the same, and it has worked there without problems for quite a while.

Reply:
This is identical to what SciPy does - I assume it works.